### ISSUE TITLE
No error for implicitly defined return value

### ISSUE THREAD

**Issue by shreydesai** [python/mypy#1869]
Correct me if this issue has already been addressed, but I found the no error for an implicitly defined return value troubling. It seems as if this would _not_ be static typing because the return value should explicitly defined (whatever it is) for statically typed languages.

I think there should be a difference between these two functions when run through the parser, in that the second one should raise an error for not having an explicitly defined return value:

```python
def f() -> None:
    print('Hello world')

def g():
    print('Hello world')
```

In statically typed languages like Java, return values must be explicitly defined, whether they are `void`, `String`s, or primitive data types, for instance:
```java
public void test1() {
  System.out.println("Hello world");
}
```

**Comment by gvanrossum** [comment]
Well, one of mypy's fundamental properties is that it doesn't complain about a function that has no annotations. So it really should not complain about the second example (g()) above.

However the following also is not considered an error, and this is indeed troubling:
```python
def f() -> int:
    print('Hw')
```
We should address this when introducing `--strict-optional` (which is currently experimental).

**Comment by gvanrossum** [comment]
See also https://github.com/python/mypy/pull/1748 which adds a new flag that *would* check for this kind of error (but still not in unannotated functions).

**Comment by rwbarton** [comment]
Perhaps the OP is more interested in something like `--disallow-untyped-defs`.

**Comment by gvanrossum** [comment]
Then again maybe this can be labeled as a question and closed?

**Comment by ddfisher** [comment]
#1748 has landed, so I'm going to close this.
